jubilant/_juju.py

```diff
@@ -997,17 +1000,41 @@ def scp(
         if isinstance(scp_options, str):
             raise TypeError('scp_options must be an iterable of str, not str')
 
-        args = ['scp']
-        if container is not None:
-            args.extend(['--container', container])
-        if not host_key_checks:
-            args.append('--no-host-key-checks')
-        args.append('--')
-        args.extend(scp_options)
-        args.append(str(source))
-        args.append(str(destination))
+        source = str(source)
+        destination = str(destination)
+        temp_needed = (':' not in source or ':' not in destination) and self._juju_is_snap
+        with (
+            tempfile.NamedTemporaryFile('w+', dir=self._temp_dir)
+            if temp_needed
+            else contextlib.nullcontext()
+        ) as file_temp:
+            args = ['scp']
+            if container is not None:
+                args.extend(['--container', container])
+            if not host_key_checks:
+                args.append('--no-host-key-checks')
+            args.append('--')
+            args.extend(scp_options)
+
+            if file_temp is not None and ':' not in source:
```

Comment:
You can `juju scp` with a local source and local destination. I'm not sure why someone would do that, but it seems like we should have it work. So I think these should be switched so that it handles adding the source arg, then handles adding the destination arg, then outside of that runs the command (and then does the destination copy if needed).